marlowe-dashboard-client/src/Template/View.purs

```diff
@@ -42,196 +45,237 @@ contractSetupScreen wallets state =
 
     templateContent = view _templateContent state
 
-    editingNickname = view _editingNickname state
+    slotContent = view _slotContent templateContent
+
+    valueContent = view _valueContent templateContent
 
-    setupProgress = view _setupProgress state
+    termsAreAccessible = roleWalletsAreValid roleWallets wallets
+
+    payIsAccessible = termsAreAccessible && templateContentIsValid templateContent
   in
     div
-      [ classNames [ "grid", "grid-rows-contract-setup", "h-full", "overflow-hidden" ] ]
+      [ classNames [ "grid", "grid-rows-contractSetup", "h-full", "overflow-hidden" ] ]
       [ div
-          [ classNames [ "border-b", "border-darkgray", "bg-grayblue" ] ]
+          [ classNames [ "flex", "justify-between", "items-center", "px-6", "py-2", "border-b", "border-darkgray" ] ]
           [ a
-              [ classNames [ "flex", "items-center", "p-2", "font-semibold" ]
+              -- "-ml-2" shifts things to the left so that the icon lines up properly
+              [ classNames [ "flex", "items-center", "font-semibold", "-ml-2" ]
               , onClick_ ToggleTemplateLibraryCard
               ]
               [ Icon.previous
               , span_
                   [ text "Choose template" ]
               ]
+          , span
+              [ classNames [ "text-sm", "uppercase" ] ]
+              [ text contractName ]
           ]
-      , contractNicknameDisplay contractName contractNickname editingNickname
+      , contractNicknameDisplay contractName contractNickname
       , div
-          [ classNames [ "h-full", "overflow-y-auto", "grid" ] ]
-          [ subHeader "top-0" Icon.roles "Roles"
-          , contractRoles wallets metaData extendedContract roleWallets
-          , subHeader "top-10" Icon.terms "Terms"
-          , contractParameters metaData templateContent
-          , subHeader "top-20" Icon.pay "Review and pay"
-          , contractReview state
+          [ classNames [] ]
```

Comment:
if this is a placeholder all good, when it's closer to final version don't forget to do div_